src/node_util.cc

```diff
@@ -254,12 +255,59 @@ static void ParseEnv(const FunctionCallbackInfo<Value>& args) {
   args.GetReturnValue().Set(dotenv.ToObject(env));
 }
 
+static void GetCallSite(const FunctionCallbackInfo<Value>& args) {
+  Environment* env = Environment::GetCurrent(args);
+  Isolate* isolate = env->isolate();
+
+  CHECK_EQ(args.Length(), 1);
+  CHECK(args[0]->IsNumber());
+  const uint32_t frames = args[0].As<Uint32>()->Value();
+
+  // +1 for disregarding node:util
+  Local<StackTrace> stack = StackTrace::CurrentStackTrace(isolate, frames + 1);
+  const int frame_count = stack->GetFrameCount();
+  LocalVector<Value> callsite_objects(isolate);
+
+  // Frame 0 is node:util. It should be skipped.
+  for (int i = 1; i < frame_count; ++i) {
+    Local<Object> obj = Object::New(isolate);
```

Comment:
non-blocking: there's a variation on `Object::New(...)` that accepts an array of keys and values to create the object more efficiently in a single call. Calling `Set` for each individual property is slightly less performant. It might not be worth the additional work tho. Worth benchmarking at some point, however.

Reply:
I'll work on that in a follow up PR. I have plans to make this API faster. I won't forget. Thanks